src/components/tools/Location/StaticMap.svelte

```diff
@@ -34,41 +31,55 @@
     };
   }
 
-  $: if (location) {
-    if (location.geometry.type === "Point") {
-      geojson = location.geometry;
-      const [lng, lat] = geojson.coordinates;
-      // Use zoom for point geometry to prevent returned image from
-      // being zoomed all the way in
-      bounds = `${lng},${lat},${zoom}`;
-      overlay = encodeURIComponent(JSON.stringify(geojson));
-      // Padding cannot be used without the auto parameter/bounding box.
-      src = `https://api.mapbox.com/styles/v1/${style}/static/geojson(${overlay})/${bounds}/${width}x${height}?&access_token=${mapboxgl.accessToken}`;
+  function createPolygonImgSrc({ geometry }) {
+    // Add style for non point geometry
+    const geojson = {
+      type: "Feature",
+      properties: {
+        "fill-opacity": 0,
+        "stroke-width": 3,
+      },
+      geometry,
+    };
+    const bounds = "auto";
+    // The Static Images API only accepts requests that are 8,192 or fewer characters long. Reduce the length of the geojson overlay or request a static map without the overlay
+    let overlay;
+    let simplifiedGeojson;
+    const geojsonString = JSON.stringify(geojson);
+
+    if (geojsonString.length < 7000) {
+      overlay = encodeURIComponent(geojsonString);
     } else {
-      // Add style for non point geometry
-      geojson = {
-        type: "Feature",
-        properties: {
-          "fill-opacity": 0,
-          "stroke-width": 3,
-        },
-        geometry: location.geometry,
-      };
-      // The Static Images API only accepts requests that are 8,192 or fewer characters long.
-      // Simplify geometry only for more complex shapes
-      const geojsonString = JSON.stringify(geojson);
-      if (geojsonString.length < 7000) {
-        overlay = encodeURIComponent(geojsonString);
-      } else {
-        const simplifiedGeojson = simplify(geojson, {
-          tolerance: 0.01,
-          highQuality: true,
+      // Simplify geometry for more complex shapes
+      simplifiedGeojson = simplify(geojson, {
+        tolerance: 0.005,
+        highQuality: true,
+      });
+      // Simplify further for even more complex shapes (e.g. state boundaries)
+      if (JSON.stringify(simplifiedGeojson).length > 8000) {
+        simplifiedGeojson = simplify(geojson, {
+          tolerance: 0.1,
+          highQuality: false,
         });
-        overlay = encodeURIComponent(JSON.stringify(simplifiedGeojson));
       }
-      bounds = "auto";
+      overlay = encodeURIComponent(JSON.stringify(simplifiedGeojson));
+    }
 
-      src = `https://api.mapbox.com/styles/v1/${style}/static/geojson(${overlay})/${bounds}/${width}x${height}?padding=${padding}&access_token=${mapboxgl.accessToken}`;
+    return `https://api.mapbox.com/styles/v1/${style}/static/geojson(${overlay})/${bounds}/${width}x${height}?padding=${padding}&access_token=${mapboxgl.accessToken}`;
+  }
+
+  $: if (location) {
```

Comment:
Just a suggestion, feel free to take it or leave it: I would simplify this by taking the handling of creating the point url and wrapping it in a function like `createPointImgSrc()` and then turn this into a `if else`

```js
$: if (location && location.geometry) handleLocation(location);

function handleLocation(feature) {
  if (feature.geometry.type === "Point") {
    createPointImgSrc(location);
  } else {
    createPolygonImgSrc(location);
  }
}
```

It would help improve readability and make it easier to follow the logic IMO by avoiding the nesting of `if`'s and using a function in one block but not the other.

You could also have a separate function that creates the URL string so that you don't have to duplicate the code for it for each geometry type.

Reply:
Good idea, fixed in https://github.com/berkeley-gif/caladapt-website-2021/pull/74/commits/182e8c7c41953a742e87da1ff4b8c54e5d2d9be6